subsys/bluetooth/host/audio/iso.c

```diff
@@ -930,6 +933,7 @@ int bt_iso_chan_disconnect(struct bt_iso_chan *chan)
 
 	if (chan->state == BT_ISO_BOUND) {
 		bt_iso_chan_set_state(chan, BT_ISO_DISCONNECTED);
+		bt_iso_chan_remove(chan->conn, chan);
 		bt_conn_unref(chan->conn);
```

Comment:
You are sure you dont' have to check `if (chan->conn) {` here as well?

Reply:
There is a check already for that above this statement.